### PR TITLE
plat-sam: add psci support for reset and shutdown

### PR DESCRIPTION
**Pull request by clementleger**
This PR adds support for reset and shutdown via PSCI for plat-sam. It depends on reset controller and shutdown controller support which is currently under review.

### CODE REVIEW

File: core/arch/arm/plat-sam/pm/psci.c [hunk 1]
```diff
@@ -3,16 +3,39 @@
  * Copyright (c) 2021, Microchip
  */
 
+#include <drivers/atmel_rstc.h>
```
Comment (etienne-lms) [inline core/arch/arm/plat-sam/pm/psci.c:6]: In commit "plat-sam: enable PSCI reset support": `console.h` should be before `drivers/xxx.h`.

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with comment on commit "plat-sam: enable PSCI reset support" addressed.

File: core/arch/arm/plat-sam/pm/psci.c (hunk 1 above)
Comment (clementleger) [reply core/arch/arm/plat-sam/pm/psci.c:6]: Even better, removed it since not needed anymore.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by clementleger** [comment]
New version:
- Added Acked-by Jens

**Pull request merged**